src/main/kotlin/org/rust/cargo/RustfmtWatcher.kt

```diff
@@ -85,8 +84,13 @@ class RustfmtWatcher : BaseComponent {
             val project = cargoProject.project
             if (!project.rustSettings.runRustfmtOnSave) return
             val rustfmt = project.toolchain?.rustfmt() ?: return
-            if (checkNeedInstallRustfmt(cargoProject.project, cargoProject.workingDirectory)) return
-            documents.forEach { rustfmt.reformatDocument(cargoProject, it) }
+
+            ApplicationManager.getApplication().invokeLater {
+                for (document in documents) {
+                    rustfmt.reformatDocument(cargoProject, document)
+                    FileDocumentManager.getInstance().saveDocument(document)
```

Comment:
With this solution the document will be saved twice. I don't sure it's a good idea